tests/test_annotated.py

```diff
@@ -111,6 +105,22 @@ class M(BaseModel):
             x: hint = value
 
 
+@pytest.mark.parametrize(
+    'metadata',
+    [
+        0,
+        'foo',
+    ],
+)
+def test_annotated_allows_unknown(metadata):
+    class M(BaseModel):
+        x: Annotated[int, metadata] = 5
+
+    field_info = M.model_fields['x']
+    assert len(field_info.metadata) == 1
+    assert metadata in field_info.metadata, 'Records the unknown metadata'
```

Comment:
we should also check that `metadata` is included in `M.__annotations__` - that's where any other library would look for is.